### DISCUSSION TITLE
github copilot to answer documentation questions -- is coming.

### DISCUSSION THREAD

**Discussion by taigit** [sqlalchemy/sqlalchemy#9545] [Ideas]
I'm looking forward to asking about sqlalchemy using AI generated answers:
https://githubnext.com/projects/copilot-for-docs/

AI will read the hard-earned, precise documentation of sqlalchemy.

I tried it in chatcpt-3.5:

I used a feature mentioned at   https://docs.sqlalchemy.org/en/20/changelog/changelog_14.html#change-1.4.21

`
As of my knowledge cutoff date of September 2021, the latest version of SQLAlchemy 1.4 was 1.4.25, which was released on August 30th, 2021.

.
.
Q: Added new method HasCTE.add_cte() -- what does it do?
.

As of my knowledge cutoff date of September 2021, the HasCTE class is not a built-in class in Python but a specific class used in the SQLAlchemy library. However, I can provide some general information on what Common Table Expressions (CTEs) are and how they are used in SQLAlchemy.

In SQLAlchemy, a CTE is a named temporary result set that you can reference within a SELECT, INSERT, UPDATE, or DELETE statement. CTEs can help simplify complex queries and make them more readable by breaking them down into smaller, reusable parts.

The HasCTE class in SQLAlchemy provides a set of methods that allow you to build queries that use CTEs. The add_cte() method is a new method added to this class that allows you to define and add a new CTE to a query. The add_cte() method takes two arguments:

name: A string that specifies the name of the CTE.
selectable: A SQLAlchemy Selectable object that defines the SELECT statement that generates the CTE.
Here's an example of how you might use the add_cte() method to define a CTE in a query:

< code is here >

In this example, we define a CTE named my_cte that selects the id and name columns from a table. We then build a query that references this CTE using the with_cte() method of the HasCTE class. Finally, we execute the query, passing in a value for the id parameter.

Note that the exact behavior of the add_cte() method may vary depending on the specific version of SQLAlchemy you are using, and that the example above is just one possible usage of this method.

`

The code it showed:

[python_code.txt](https://github.com/sqlalchemy/sqlalchemy/files/11069705/python_code.txt)

**Comment by CaselIT** [comment]
Hi,

Yes, mike and me signed up for the wait list. Once it's enabled we will see what i can do.
I don't you if you can point it to an external site, at least initially, so we will see I guess
